src/hooks/index.ts

```diff
@@ -40,15 +40,25 @@ export function useEagerConnect() {
  * and out after checking what network theyre on
  */
 export function useInactiveListener(suppress = false) {
-    const { active, error, activate } = useWeb3ReactCore() // specifically using useWeb3React because of what this hook does
+    const { active, error, activate, deactivate } = useWeb3ReactCore() // specifically using useWeb3React because of what this hook does
 
     useEffect(() => {
         const { ethereum } = window
 
+        if (window.ethereum) {
+          let provider:any
+          const isMultiple = window.ethereum?.providers?.length > 1
+          !isMultiple ?  provider = ethereum : 
+                         provider = window.ethereum?.providers.find((isMetaMask: any) => isMetaMask.isMetaMask) 
+          window.ethereum.selectedProvider = provider  
+        }
+
         if (ethereum && ethereum.on && !active && !error && !suppress) {
```

Comment:
should this be changed to ethereum.selectedProvider.on?

Reply:
the .on does not always seem to live there
I spent quite a bit of time on this

Tried numerous different ways, (checking both, etc)

for some reason, this is the only one that seems to work in all three cases